x-pack/plugins/security_solution/public/timelines/components/timeline/body/column_headers/helpers.ts

```diff
@@ -23,7 +39,13 @@ export const getColumnHeaders = (
         return {
           ...header,
           ...get(
-            [splitHeader.length > 1 ? splitHeader[0] : 'base', 'fields', header.id],
+            [
+              splitHeader.length > 1
+                ? splitHeader[0]
+                : getRootCategory({ field: header.id, browserFields }),
+              'fields',
+              header.id,
+            ],
```

Comment:
While we're touching this part of the code, it might also be a good idea to not inline this piece. It's rather complex to make sense of like this imo. Maybe extracting this into its own variable before the return will help. Wdyt?

Reply:
Agreed! Updated [here](https://github.com/elastic/kibana/pull/132570/commits/606927c560ff0fe8ad2e83998171ea9a4bac8c7e#diff-f48e2df5848c4cc81395df0ed8f29bb03652cc4c66fb0fe76cbc50429a6f1301R38)